objective.py

```diff
@@ -29,63 +38,30 @@ def set_data(self, X, y):
         self.X, self.y = X, y
         self.n_samples, self.n_features = X.shape
 
-        # the following code fits the path once using the same path-stopping
-        # criteria that glmnet uses to find a reasonable lambda path in order
-        # to obtain a reasonable lambda sequence
-        if self.lambda_min_ratio is None:
-            self.lambda_min_ratio = 1e-2 if self.n_samples < self.n_features else 1e-4
-
-        lambda_max = self._get_lambda_max()
-        lambdas = np.geomspace(
-            lambda_max, lambda_max * self.lambda_min_ratio, self.n_lambda
-        )
-
-        # the pmax setting is redundant because dfmax is hardcoded, but it's
-        # kept just to be explicit about how this is done
-        dfmax = self.n_features + 1
-        pmax = min(dfmax * 2 + 20, self.n_features)
-
-        lasso = Lasso(
-            max_iter=10000,
-            alpha=lambda_max / self.n_samples,
-            # TODO(jolars): consider whether tol might be too high
-            tol=1e-4,
-            fit_intercept=self.fit_intercept,
-            warm_start=True,
+        # run glmnet once to obtain a lambda sequence
+        if sparse.issparse(X):
+            r_Matrix = packages.importr("Matrix")
+            X_coo = X.tocoo()
+            X_tmp = r_Matrix.sparseMatrix(
+                i=robjects.IntVector(X_coo.row + 1),
+                j=robjects.IntVector(X_coo.col + 1),
+                x=robjects.FloatVector(X_coo.data),
+                dims=robjects.IntVector(X_coo.shape),
+            )
+        else:
+            X_tmp = robjects.r.matrix(X, X.shape[0], X.shape[1])
+
+        y_tmp = robjects.FloatVector(y)
+        glmnet = robjects.r["glmnet"]
+        glmnet_fit = glmnet(
+            X_tmp,
+            y_tmp,
+            intercept=self.fit_intercept,
+            standardize=False,
+            maxit=1_000_000,
         )
```

Comment:
don't you have the value of the results also with this call to `glmnet`? it would be nice to add the computation of the iterates convergence with these?

I would advise to take this computation in a standalone function, and use a caching mechanism to call this? We can discuss about this during the sprint if you want.